crml/cennzx-spot/src/lib.rs

```diff
@@ -0,0 +1,858 @@
+// Copyright (C) 2019 Centrality Investments Limited
+// This file is part of CENNZnet.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//!
+//! CENNZX-Spot exchange
+//!
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[macro_use]
+mod tests;
+mod impls;
+mod types;
+pub use impls::{ExchangeAddressFor, ExchangeAddressGenerator};
+pub use types::{FeeRate, U256};
+
+#[macro_use]
+extern crate support;
+
+use generic_asset;
+use rstd::prelude::*;
+use runtime_primitives::traits::{Bounded, One, Zero};
+use support::{
+	dispatch::{Dispatchable, Result as DispatchResult},
+	Parameter, StorageDoubleMap, StorageMap, StorageValue,
+};
+use system::{ensure_root, ensure_signed};
+
+// TODO: remove this temp fix for overflow issue when upstream fix ready
```

Comment:
Looks like this is ok to remove. I think there were a few iterations of "temp fix" and the comments were not updated